src/MultiFormats/MultiHash.cs

```diff
@@ -0,0 +1,570 @@
+
+        /// <summary>
+        ///   Determines if the identity hash algorithm is in use.
+        /// </summary>
+        /// <value>
+        ///   <b>true</b> if the identity hash algorithm is used; otherwise, <b>false</b>.
+        /// </value>
+        /// <remarks>
+        ///   The identity hash is used to inline a small amount of data into a <see>
+        ///       <cref>Lib.P2P.Cid</cref>
+        ///   </see>
+        ///   .
+        ///   When <b>true</b>, the <see cref="Digest"/> is also the content.
+        /// </remarks>
+        public bool IsIdentityHash => Algorithm.Code == 0;
+
+        /// <summary>
+        ///   Writes the binary representation of the multihash to the specified <see cref="Stream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="Stream"/> to write to.
+        /// </param>
+        /// <remarks>
+        ///   The binary representation is a 1-byte <see cref="HashingAlgorithm.Code"/>,
+        ///   1-byte <see cref="HashingAlgorithm.DigestSize"/> followed by the <see cref="Digest"/>.
+        /// </remarks>
+        public void Write(Stream stream)
+        {
+            using (var cos = new CodedOutputStream(stream, true))
+            {
+                Write(cos);
+            }
+        }
+
+        /// <summary>
+        ///   Writes the binary representation of the multihash to the specified <see cref="Google.Protobuf.CodedOutputStream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="Google.Protobuf.CodedOutputStream"/> to write to.
+        /// </param>
+        /// <remarks>
+        ///   The binary representation is a <see cref="Varint"/> of the <see cref="HashingAlgorithm.Code"/>,
+        ///   <see cref="Varint"/> of the <see cref="HashingAlgorithm.DigestSize"/> followed by the <see cref="Digest"/>.
+        /// </remarks>
+        public void Write(CodedOutputStream stream)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            stream.WriteInt32(Algorithm.Code);
+            stream.WriteLength(Digest.Length);
+            stream.WriteSomeBytes(Digest);
+        }
+
+        private void Read(Stream stream)
+        {
+            using (var cis = new CodedInputStream(stream, true))
+            {
+                Read(cis);
+            }
+        }
+
+        private void Read(CodedInputStream stream)
+        {
+            var code = stream.ReadInt32();
+            var digestSize = stream.ReadLength();
+
+            HashingAlgorithm.Codes.TryGetValue(code, out var a);
+            Algorithm = a;
+            if (Algorithm == null)
+            {
+                Algorithm = HashingAlgorithm.Register("ipfs-" + code, code, digestSize);
+                RaiseUnknownHashingAlgorithm(Algorithm);
+            }
+            else if (Algorithm.DigestSize != 0 && digestSize != Algorithm.DigestSize)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add the missing 'else' clause.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)